seqr/views/apis/saved_variant_api.py

```diff
@@ -72,110 +61,180 @@ def create_saved_variant_handler(request):
 
-    variant_note = {variant_guid: {
-        'notes': [get_json_for_variant_note(tag) for tag in saved_variant.variantnote_set.all()]},
-    }
+        saved_variants.append(saved_variant)
+
+    _create_variant_note(saved_variants, request_json, request.user)
+
+    variant_note = {}
+    for variant_guid in variant_guids:
+        variant_note[variant_guid] = {'notes': [get_json_for_variant_note(tag) for tag in saved_variant.variantnote_set.all()]}
 
     return create_json_response({
         'savedVariantsByGuid': variant_note,
         'genesById': gene_note
     })
 
 
-def _create_variant_note(saved_variant, note_json, user):
+def _create_variant_note(saved_variants, note_json, user):
     create_seqr_model(
         VariantNote,
-        saved_variant=saved_variant,
         note=note_json.get('note'),
         submit_to_clinvar=note_json.get('submitToClinvar') or False,
         search_hash=note_json.get('searchHash'),
         created_by=user,
-    )
+    ).saved_variants.add(*saved_variants)
 
 
 @login_required(login_url=API_LOGIN_REQUIRED_URL)
 @csrf_exempt
-def update_variant_note_handler(request, variant_guid, note_guid):
-    saved_variant = SavedVariant.objects.get(guid=variant_guid)
-    check_permissions(saved_variant.family.project, request.user, CAN_VIEW)
-    note = VariantNote.objects.get(guid=note_guid, saved_variant=saved_variant)
+def update_variant_note_handler(request, variant_guids, note_guid):
+    variant_guids = variant_guids.split(',')
+    saved_variants = []
+    for variant_guid in variant_guids:
+        saved_variant = SavedVariant.objects.get(guid=variant_guid)
+        check_permissions(saved_variant.family.project, request.user, CAN_VIEW)
+        saved_variants.append(saved_variant)
+    note = VariantNote.objects.get(guid=note_guid)
 
     request_json = json.loads(request.body)
     update_model_from_json(note, request_json, allow_unknown_keys=True)
 
-    return create_json_response({'savedVariantsByGuid': {variant_guid: {
-        'notes': [get_json_for_variant_note(tag) for tag in saved_variant.variantnote_set.all()]
-    }}})
+    update = {}
+    for variant_guid in variant_guids:
+        update[variant_guid] = {
+            'notes': [get_json_for_variant_note(note) for note in saved_variant.variantnote_set.all()],
+        }
+    return create_json_response({'savedVariantsByGuid': update})
 
 
 @login_required(login_url=API_LOGIN_REQUIRED_URL)
 @csrf_exempt
-def delete_variant_note_handler(request, variant_guid, note_guid):
-    saved_variant = SavedVariant.objects.get(guid=variant_guid)
-    check_permissions(saved_variant.family.project, request.user, CAN_VIEW)
-    note = VariantNote.objects.get(guid=note_guid, saved_variant=saved_variant)
+def delete_variant_note_handler(request, variant_guids, note_guid):
+    variant_guids = variant_guids.split(',')
+    saved_variants = []
```

Comment:
You can do this as one query with `saved_variants = SavedVariant.objects.get(guid__in= variant_guids)`